src/commands/selection.rs

```diff
@@ -6,32 +6,47 @@ use crate::commands::{self, Result};
 use crate::util;
 
 pub fn delete(app: &mut Application) -> Result {
-    if let Some(buffer) = app.workspace.current_buffer() {
-        match app.mode {
-            Mode::Select(ref select_mode) => {
-                let cursor_position = *buffer.cursor.clone();
-                let delete_range = Range::new(cursor_position, select_mode.anchor);
-                buffer.delete_range(delete_range.clone());
-                buffer.cursor.move_to(delete_range.start());
-            }
-            Mode::SelectLine(ref mode) => {
-                let delete_range = mode.to_range(&*buffer.cursor);
-                buffer.delete_range(delete_range.clone());
-                buffer.cursor.move_to(delete_range.start());
-            }
-            Mode::Search(ref mode) => {
-                let selection = mode.results
-                    .as_ref()
-                    .and_then(|r| r.selection())
-                    .ok_or("Can't delete in search mode without a selected result")?;
-                buffer.delete_range(selection.clone());
-            }
-            _ => bail!("Can't delete selections outside of select mode"),
-        };
-    } else {
+    if app.workspace.current_buffer().is_none() {
+        bail!(BUFFER_MISSING);
+    }
+    match app.mode {
+        Mode::Select(_) | Mode::SelectLine(_) | Mode::Search(_) => {
+            let delete_range = range_from(app)?;
+            let buffer = app.workspace.current_buffer().unwrap();
+
+            buffer.delete_range(delete_range.clone());
+            buffer.cursor.move_to(delete_range.start());
+        }
+        _ => bail!("Can't delete selections outside of select mode"),
+    };
+
+    Ok(())
+}
+
+pub fn justify(app: &mut Application) -> Result {
+    if app.workspace.current_buffer().is_none() {
         bail!(BUFFER_MISSING);
     }
 
+    let range = range_from(app)?;
+
+    // delete and save the range, then justify that range
+    let path = &app.workspace.path.clone();
+    let buffer = app.workspace.current_buffer().unwrap();
+
+    if let Some(text) = buffer.read(&range.clone()) {
+        buffer.delete_range(range.clone());
+        buffer.cursor.move_to(range.start());
+
+        buffer.insert(
+            &justify_string(
+                &text,
+                app.preferences.borrow().line_length_guide().unwrap_or(80),
```

Comment:
We already specify a default value in the preferences file itself. If someone has explicitly disabled the line length guide, let's raise an error here:

```suggestion
                app.preferences.borrow().line_length_guide().ok_or("Cannot justify without line length guide")
```